src/modules/hip/kernel/log.hpp

```diff
@@ -0,0 +1,114 @@
+#include <hip/hip_runtime.h>
+#include "rpp_hip_common.hpp"
+
+template <typename T>
+__device__ void log_hip_compute(T *srcPtr, d_float8 *src_f8, d_float8 *dst_f8)
+{
+    if constexpr (std::is_same<T, schar>::value)
+        rpp_hip_math_add8_const(src_f8, src_f8, (float4)128);
+
+    dst_f8->f1[0] = __logf(fabsf(src_f8->f1[0]));
+    dst_f8->f1[1] = __logf(fabsf(src_f8->f1[1]));
+    dst_f8->f1[2] = __logf(fabsf(src_f8->f1[2]));
+    dst_f8->f1[3] = __logf(fabsf(src_f8->f1[3]));
+    dst_f8->f1[4] = __logf(fabsf(src_f8->f1[4]));
+    dst_f8->f1[5] = __logf(fabsf(src_f8->f1[5]));
+    dst_f8->f1[6] = __logf(fabsf(src_f8->f1[6]));
+    dst_f8->f1[7] = __logf(fabsf(src_f8->f1[7]));
+
+    if constexpr (std::is_same<T, schar>::value)
+    {
+        dst_f8->f4[0] = rpp_hip_pixel_check_0to255(dst_f8->f4[0]) - (float4)128;
+        dst_f8->f4[1] = rpp_hip_pixel_check_0to255(dst_f8->f4[1]) - (float4)128;
+    }
+}
+
+// F16 stores without layout toggle (8 F16 pixels)
+
+__device__ __forceinline__ void rpp_hip_pack_float8_and_store(half *dstPtr, d_float8 *dst_f8)
+{
+    d_half8 dst_h8;
+
+    dst_h8.h2[0] = __float22half2_rn(make_float2(dst_f8->f1[0], dst_f8->f1[1]));
+    dst_h8.h2[1] = __float22half2_rn(make_float2(dst_f8->f1[2], dst_f8->f1[3]));
+    dst_h8.h2[2] = __float22half2_rn(make_float2(dst_f8->f1[4], dst_f8->f1[5]));
+    dst_h8.h2[3] = __float22half2_rn(make_float2(dst_f8->f1[6], dst_f8->f1[7]));
+
+    *(d_half8 *)dstPtr = dst_h8;
+}
+
+// F32 stores without layout toggle (8 F32 pixels)
+
+template <typename T>
+__device__ __forceinline__ void rpp_hip_pack_float8_and_store(T *dstPtr, d_float8 *dst_f8)
+{
+    *(d_float8_s *)dstPtr = *(d_float8_s *)dst_f8;
+}
+
+template <typename T1, typename T2>
+__global__ void log_generic_hip_tensor(T1 *srcPtr,
+                                       uint *srcStrides,
+                                       uint *srcDims,
+                                       uint srcNumDims,
+                                       T2 *dstPtr,
+                                       uint *dstStrides,
+                                       Rpp32u *roiTensor)
+{
+    int id_x = (hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x) * 8;
+    int id_y = hipBlockIdx_y * hipBlockDim_y + hipThreadIdx_y;
+
+    if(id_x >= srcStrides[0])
+        return;
+
+    uint *roi = &roiTensor[id_y * srcNumDims * 2 + srcNumDims];
+    uint dstIdx = (id_y * *dstStrides++);
+    uint srcIdx = (id_y * *srcStrides++);
+    uint coords[RPPT_MAX_DIMS];
+
+    for (int i = 0; i < srcNumDims; i++)
+    {
+        coords[i] = (id_x / srcStrides[i]) % srcDims[i];
+        if(coords[i] > roi[i])
```

Comment:
Instead of this coords compute which is very costly (since mod is involved)
You can think of having of separate kernels for 1D, 2D, 3D data and process them without this compute

Just to understand the performance impact this has, you can create a separate 2D kernel and compare it's performance with the current generic kernel for all dims

@r-abishek 
Please share your thoughts here

Reply:
@sampath1117 @snehaa8 Yes, for increased dimensions (upto 2/3/4), or for filter type kernels with 3/5/7/9 filter-sizes in hip, just copy relevant code to a separate smaller kernel and check performance difference, if it gives you 10-15% better, its fine to separate them

Reply:
Yes, its giving minor improvement like 5% so added separate calls